### PR TITLE
Disable dragging of plots with any interactions enabled

### LINKED ISSUE #1393: Should disable image dragging if any plot interactions are enabled
If just clicking is enabled, it's distracting and annoying if you click and drag a little bit and the browser thinks you want to drag the image to a folder. (Currently, when brushing is enabled, the image dragging is disabled.)

### LINKED ISSUE #2223: Plot interaction: when click is enabled, dragging image should be disabled

### PR DESCRIPTION
**Pull request by wch**
Closes #1393, closes #2223.

Test app:

```R
library(shiny)

shinyApp(
  ui = fluidPage(
    p("The plot below doesn't have any interactions. It should be draggable."),
    plotOutput("plot_no_interact", height = 250, width = 250),
    
    p("The plot below allows click interactions. It should not be draggable."),
    plotOutput("plot_interact", height = 250, width = 250, hover = "plot_click")
  ),
  server = function(input, output) {
    output$plot_no_interact <- renderPlot({
      plot(cars)
    })
    output$plot_interact <- renderPlot({
      plot(cars)
    })
  }
)
```

### CODE REVIEW

**Comment by wch** [comment]
Note that the generated JS resources aren't part of this PR, so they will need to be rebuilt after merging.

**Comment by jcheng5** [comment]
There's a `"draggable"` attribute for img, can we just add that in R?

**Comment by wch** [comment]
We can't do it from R because the `img` tag is created in JavaScript. We could do `$img.attr('draggable', false)` in JS, but then it doesn't quite work right in Firefox -- the image can be selected.



And apparently if you try to disable selection in Firefox, then `draggable="false"` stops working: https://bugzilla.mozilla.org/show_bug.cgi?id=1376369

I seem to recall trying a bunch of different things when this code was originally written and settled on what's here. We know it works so I'd prefer to stick with it.

File: srcjs/output_binding_image.js [hunk 1]
```diff
@@ -225,6 +225,18 @@ outputBindings.register(imageOutputBinding, 'shiny.imageOutput');
...
+  $img.css('-webkit-user-drag', 'none');
+
+  // Firefox, IE<=10
+  // First remove existing handler so we don't keep adding handlers.
+  $img.off('dragstart.image_output');
+  $img.on('dragstart.image_output', function() { return false; });
+
+  // Disable selection of image and text when dragging in IE<=10
+  $el.on('selectstart.image_output', function() { return false; });
+};
```
Comment (cpsievert) [inline srcjs/output_binding_image.js]: Is this `$el` defined? `yarn build` gives me 

```
../srcjs/output_binding_image.js
  241:3  warning  '$el' is not defined  no-undef

✖ 1 problem (0 errors, 1 warning)
```

Comment (wch) [reply srcjs/output_binding_image.js]: Shoot, I copied and pasted that part and forgot to change it. I'll fix it.

File: srcjs/output_binding_image.js [hunk 2]
```diff
@@ -162,6 +166,8 @@ $.extend(imageOutputBinding, {
       }
 
       if (opts.hoverId) {
+        imageutils.disableDrag($img);
```
Comment (jcheng5) [inline srcjs/output_binding_image.js]: Did you mean for hover to disable drag?

Comment (jcheng5) [reply srcjs/output_binding_image.js]: (It's fine if you did)

File: srcjs/output_binding_image.js [hunk 3]
```diff
@@ -225,6 +225,18 @@ outputBindings.register(imageOutputBinding, 'shiny.imageOutput');
...
+  // Make image non-draggable (Chrome, Safari)
+  $img.css('-webkit-user-drag', 'none');
+
+  // Firefox, IE<=10
+  // First remove existing handler so we don't keep adding handlers.
+  $img.off('dragstart.image_output');
+  $img.on('dragstart.image_output', function() { return false; });
+
+  // Disable selection of image and text when dragging in IE<=10
+  $el.on('selectstart.image_output', function() { return false; });
```
Comment (jcheng5) [inline srcjs/output_binding_image.js]: Do you also need to remove this handler first, the way you did for `dragstart.image_output`?

Comment (wch) [reply srcjs/output_binding_image.js]: Yup, I was being sloppy with this part :(

**Comment by cpsievert** [comment]
What's the argument for having draggable plot images in the first place? Is it so that they can be drag-and-dropped into some other consumer widget? Are there other use cases? 

Also, what's the argument for disabling drag for any interaction mode? It's seems possible that someone might want draggability if hover and/or click are enabled...

**Comment by wch** [comment]
There's no particular reason for having draggable plots; that's just how images in general work in web browsers, so for the regular non-interactive case, it makes sense to just keep that behavior. The typical use case is that users can drag and drop to a file outside of the browser.

The reason for disabling drag when interactions are enabled is that when clicking is enabled, people often click and just drag without thinking about it (I do this) and then you end up dragging the image, which just gets in the way. When dragging is disabled, if users want to save the file, it's still possible by right-clicking and saving.

**Comment by cpsievert** [comment]
Can't figure how to officially approve, but LGTM 👍

**Pull request merged**